Cleanup documentation on deploying from your laptop

yuvipanda opened the pull request:
Ref https://github.com/2i2c-org/pilot-hubs/issues/351

damianavila reviewed (approved):
LGTM, this is great @yuvipanda! Thanks for adding those details, it is pretty clear now!

damianavila commented:
Just a general related question/comment (maybe silly and minimal... and this is not a blocker for merging this PR).
But having the script called `deploy` with the same name as the most important subcommand `deploy` seems prone to confuse people, IMHO (ie. Is this a copypaste error in the command?)

   ```bash
   python3 deploy deploy <cluster-name> <hub-name>
   ```

Have you discussed before about, maybe, renaming the script to something such as `deployer` (or whatever name you want except for `deploy` 😉 ).

yuvipanda commented:
@damianavila to avoid confusion, I've now renamed the `deploy` directory, so to deploy you will now have to do:

```bash
python3 python3 deploy 2i2c staging
```

I hope this is less confusing?

choldgraf commented:
is the 2 `python3`s intentional?

damianavila commented:
> is the 2 python3s intentional?

Je... maybe @yuvipanda copy-pasted the wrong word or I am bad at understanding jokes 😜

yuvipanda commented:
@damianavila @choldgraf it was, indeed, a joke :D Instead of deploy being repeated twice because it's the name of the dir and subcommand, I got python3 to repeat twice by renaming the dir to python3 :) 

Now I've done something *useful* instead and taken @damianavila's naming suggestion - the dir is now called `deployer`.

choldgraf commented:
lol @yuvipanda that was a solid troll

so the new command would be

```
python3 deployer deploy 2i2c staging
```

yuvipanda commented:
@choldgraf yes that's correct! I updated the docs too

choldgraf commented on docs/howto/operate/manual-deploy.md:
````diff
@@ -1,52 +1,74 @@
-python3 deploy.py deploy --help
-```
-Will provide information about the `deploy.py` deploy options:
+Our deployment scripts live in the [`deploy/`](https://github.com/2i2c-org/pilot-hubs/blob/master/deploy.py)
````
```suggestion
Our deployment scripts live in the [`deploy/` folder](https://github.com/2i2c-org/pilot-hubs/blob/master/deploy.py)
```

damianavila replied:
That URL link should be https://github.com/2i2c-org/pilot-hubs/blob/master/deployer with the latest change.

yuvipanda replied:
Done!

choldgraf commented on docs/howto/operate/manual-deploy.md:
````diff
@@ -1,52 +1,74 @@
+   ```
 
-    Deploys only the hub passed as argument. 
+3. Install the [Google Cloud SDK](https://cloud.google.com/sdk) so
````
What about for hubs that are on AWS or Azure? If those aren't supported right now, we should add a `note:` block to make it clear

yuvipanda replied:
I've updated the text here to clarify.

choldgraf reviewed (commented):
Just two quick comments, but this looks good to me!

damianavila commented on docs/howto/operate/manual-deploy.md:
````diff
@@ -1,52 +1,74 @@
+2. Deploy just a single hub:
+
+   ```bash
+   python3 deployer deploy <cluster-name> <hub-name>
````
❤️

yuvipanda commented:
@damianavila @choldgraf I think I've addressed your comments. I've also removed other references to deploy.py I had missed earlier, and merged contents of test-hub.md into manual-deploy.md

damianavila commented:
Super, thanks for the latest commits, LGTM again.

GeorgianaElena commented on docs/howto/configure/update-env.md:
```diff
@@ -13,7 +13,7 @@ have to manually build & push it after making a change.
 2. Make the changes you need to make to the environment, and git commit it.
 
-3. Run `python3 deploy.py build`. This will build the image and push it to
+3. Run `python3 deployer build`. This will build the image and push it to
```
I think the build command needs to be passed the cluster-name too.

damianavila replied:
That is the recent merge #358, right?

GeorgianaElena replied:
yes

yuvipanda replied:
Good catch, fixed.

GeorgianaElena reviewed (approved):
LGTM :rocket:

P.S. I love the new **deployer** name :D

yuvipanda commented:
ok, I've rebased to resolve the merge conflicts now :)

choldgraf commented:
wooo thanks for the discussion all!
